app/src/main/java/com/rai/videocards/viewmodels/VideoCardViewModel.kt

```diff
@@ -0,0 +1,96 @@
+package com.rai.videocards.viewmodels
+
+
+import androidx.lifecycle.*
+import com.rai.videocards.data.VideoCard
+import com.rai.videocards.data.VideoCardDao
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.flatMapConcat
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.launch
+
+class VideoCardViewModel(private val videoCardDao: VideoCardDao) : ViewModel()  {
+
+    val allVideoCards: LiveData<List<VideoCard>> = videoCardDao.getAll().asLiveData()
+
+
+    val searchQuery  = MutableStateFlow("")
+    private val findVideoCardsFlow = searchQuery.flatMapLatest {
+        videoCardDao.findVideoCard(it)
+    }
+    val findVideoCards = findVideoCardsFlow.asLiveData()
+
+    fun retrieveVideoCard(id: Int): LiveData<VideoCard> {
+        return videoCardDao.getVideoCard(id).asLiveData()
+    }
+
+    private fun insertVideoCard(videoCard: VideoCard) {
+        viewModelScope.launch {
+            videoCardDao.insert(videoCard)
+        }
+    }
+
+    private fun updateVideoCard(videoCard: VideoCard) {
+        viewModelScope.launch {
+            videoCardDao.update(videoCard)
+        }
+    }
+
+
+    private fun getUpdatedItemEntry(
+        id: Int, gpuModel: String, chipsetManufacturer: String, brand: String, memorySize: Int, memoryType: String
+    ): VideoCard {
+        return VideoCard(
+            id = id,
+            gpuModel = gpuModel,
+            chipsetManufacturer = chipsetManufacturer,
+            brand = brand,
+            memorySize = memorySize,
+            memoryType = memoryType
+
+        )
+    }
+
+    fun updateItem(
+        id: Int, gpuModel: String, chipsetManufacturer: String, brand: String, memorySize: Int, memoryType: String
+    ) {
```

Comment:
лучше сразу использовать DTO,  а не передавать полями
```suggestion
    fun updateItem(videoCard: VideoCard) {
```